pkg/utils/kvauth/kvauth.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package kvauth
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/joomcode/errorx"
+
+	"github.com/gtank/cryptopasta"
+	"go.etcd.io/etcd/clientv3"
+)
+
+const etcdKvAuthKeyPath = "/dashboard/kv_auth"
+
+var (
+	ErrNS               = errorx.NewNamespace("error.kvauth")
+	ErrAccountNotFound  = ErrNS.NewType("account_not_found")
+	ErrPasswordNotMatch = ErrNS.NewType("password_not_match")
+)
+
+type Auth struct {
+	Username string `json:"username"`
+	Password string `json:"password"`
+}
+
+// RevokeKvAuthKey delete the etcd path of KV mode user account
```

Comment:
The function comment should mainly describe its user oriented behaviour, instead of the internal implementation. A better one could be:

```
// RevokeKvAuthKey revokes the KV mode user account stored in etcd.
```